annotation-utils/src/main/java/edu/isi/vista/annotationutils/AceCorpusTextSource.java

```diff
@@ -0,0 +1,52 @@
+package edu.isi.vista.annotationutils;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+import edu.isi.nlp.symbols.Symbol;
+import com.google.common.base.Optional;
+import edu.isi.nlp.io.OriginalTextSource;
+import java.io.File;
+import java.nio.file.Path;
+import java.io.IOException;
+
+
+public final class AceCorpusTextSource implements OriginalTextSource{
+    private  File aceDataDirectory;
+    private File[] sourceFileDirectories;
+
+    public Optional<String> getOriginalText(final Symbol docID) throws IOException {
+        String rawSourceText = "";
+        // Search and find file in the source directories
+        for (File sourceDir : sourceFileDirectories){
+            File sourceDocFile = new File(sourceDir, docID.toString() + ".sgm");
+            if(sourceDocFile.exists()) {
+                rawSourceText = Files.asCharSource(sourceDocFile, Charsets.UTF_8).read();
+                break;
+            }
+        }
+        if(rawSourceText.isEmpty()) {
+            throw new IOException("Source file is empty for document: " + docID.toString());
+        }
+        return Optional.of(processAceSourceText(rawSourceText));
+    }
+
+    AceCorpusTextSource(File aceDataDirectory) throws IOException{

```

Comment:
style: You don't want your constructors to do any work. Prefer writing a private constructor which simply assigns its arguments to the fields. Then shift the logic from here into a `public static AceCorpusTextSource fromDirectory(....)` factory function.

Homework assignment: read, mark, learn, and inwardly digest Joshua Bloch's _Effective Java_. :-)